astropy/table/table.py

```diff
@@ -335,6 +353,203 @@ def __get__(self, instance, owner_cls):
+
+    def __repr__(self):
+        if hasattr(self, '_instance_ref'):
+            out = f'<{self.__class__.__name__} name={self.name} value={self()}>'
+        else:
+            out = super().__repr__()
+        return out
+
+    def _add_remove_setup(self, names):
+        """Common setup for add and remove.
+
+        - Coerce attribute value to a list
+        - Coerce names into a list
+        - Get the parent table instance
+        """
+        names = [names] if isinstance(names, str) else list(names)
+        # Get the value. This is the same as self() but we need `instance` here.
+        instance = self._instance_ref()
+        value = super().__get__(instance, instance.__class__)
+        value = [] if value is None else list(value)
+        return instance, names, value
+
+    def add(self, names):
+        """Add ``names`` to the include/exclude attribute.
+
+        Parameters
+        ----------
+        names : str, list, tuple
+            Column name(s) to add
+        """
+        instance, names, value = self._add_remove_setup(names)
+        value.extend(name for name in names if name not in value)
+        super().__set__(instance, tuple(value))
+
+    def remove(self, names):
+        """Remove ``names`` from the include/exclude attribute.
+
+        Parameters
+        ----------
+        names : str, list, tuple
+            Column name(s) to remove
+        """
+        self._remove(names, raise_exc=True)
+
+    def _remove(self, names, raise_exc=False):
+        """Remove ``names`` without checking if they exist"""
+        instance, names, value = self._add_remove_setup(names)
+
+        # Return now if there are no attributes and thus no action to be taken.
+        if raise_exc is False and '__attributes__' not in instance.meta:
+            return
+
+        # Remove one by one, optionally raising an exception if name is missing.
+        for name in names:
+            if name in value:
+                value.remove(name)  # Using the list.remove method
+            elif raise_exc:
+                raise ValueError(f'{name} not in {self.name}')
+
+        # Change to either None or a tuple for storing back to attribute
+        value = None if value == [] else tuple(value)
+        self.__set__(instance, value)
+
+    def _rename(self, name, new_name):
+        """Rename ``name`` to ``new_name`` if ``name`` is in the list"""
+        names = self() or ()
+        if name in names:
+            new_names = list(names)
+            new_names[new_names.index(name)] = new_name
+            self.set(new_names)
+
+    def set(self, names):
+        """Set value of include/exclude attribute to ``names``.
+
+        Parameters
+        ----------
+        names : None, str, list, tuple
+            Column name(s) to store, or None to clear
+        """
+        instance = self._instance_ref()
+        self.__set__(instance, names)
+
+    @contextlib.contextmanager
```

Comment:
Just a thought: `astropy.utils.state.ScienceState.set()` returns a context so that it can be used as a context manager if need be. I can see the benefit of just sticking with the `ConfigItem` analogy, though.

Reply:
Good idea with the `ConfigItem` analogy. I had to convince myself that the dual setting and context manager functionality is bombproof, but it seems to be. I like the API somewhat better.